src/utils.ts

```diff
@@ -1344,3 +1344,56 @@ export async function fileIsAccessible(fileName: string, mode?: number) {
     return false;
   }
 }
+
+/** @internal */
+export class MovingWindow {
+  /** Index of the next slot to be overwritten */
+  private writeIndex: number;
+  length: number;
+  samples: Float64Array;
+
+  constructor(windowSize = 10) {
+    this.samples = new Float64Array(windowSize);
+    this.length = 0;
+    this.writeIndex = 0;
+  }
+
+  addSample(sample: number) {
```

Comment:
Can you add documentation for ourselves about the expectations of this and the other methods? (specifically the circular buffer / window logic)